invokeai/backend/model_manager/util/model_util.py

```diff
@@ -133,3 +133,16 @@ def _get_shape_1(key: str, tensor: torch.Tensor, checkpoint: Dict[str, torch.Ten
             break
 
     return lora_token_vector_length
+
+
+def convert_bundle_to_flux_transformer_checkpoint(
+    transformer_state_dict: dict[str, torch.Tensor],
+) -> dict[str, torch.Tensor]:
+    original_state_dict: dict[str, torch.Tensor] = {}
+    for k, v in transformer_state_dict.items():
+        if not k.startswith("model.diffusion_model"):
+            continue
+        if k.endswith("scale"):
+            v = v.to(dtype=torch.bfloat16)
```

Comment:
Add a comment explaining why we specifically cast the scales.

Reply:
done